### ISSUE TITLE
[Outlook Mac]: Empty ItemId in pinned Taskpane without ItemChanged-Handler

### ISSUE THREAD

**Issue by Flyingdot** [OfficeDev/office-js#4892]
# Provide required information needed to triage your issue
We are reloading our Add-In in `SelectedItemsChanged` event - regardless if the Taskpane is pinned or not.
On Mac we are facing a very strange behavior. If the Taskpane is pinned, and we do not register a handler for the ItemChanged-Handler, the ItemId is empty afterwards. If we do register a handler, even an empty one, the ItemId is available.

## Your Environment

* Platform: Mac
* Host: Outlook
* Office version number: 16.89 (24090815)
* Operating System: MacOS 14.6.1
 
## Expected behavior
ItemId is available

## Current behavior
The ItemId is only available, if we do register an ItemChanged Eventhandler.

## Steps to reproduce
Create a simple AddIn using the HTML in the Code sample.

1. Open the Add-In taskpane
2. Pin the taskpane
3. Change E-Mail selection a few times

### Not working sample
```html
<!doctype html>
<html lang="en">
    <head>
        <meta charset="UTF-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
        <title>ItemId Bug</title>
        <script src="https://appsforoffice.microsoft.com/lib/1.1/hosted/office.js"></script>
    </head>
    <body class="bg-slate-100">
        <div id="out"></div>

        <script>
            Office.onReady(() => {
                const itemId = Office.context.mailbox.item.itemId
                document.getElementById('out').innerText = itemId

                Office.context.mailbox.addHandlerAsync(
                    Office.EventType.SelectedItemsChanged,
                    () => window.location.reload(),
                    function (result) {
                        if (result.status !== Office.AsyncResultStatus.Succeeded) {
                            console.error(result.error.message)
                        }
                    }
                )
            })
        </script>
    </body>
</html>
```

### working sample
```html
<!doctype html>
<html lang="en">
    <head>
        <meta charset="UTF-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
        <title>ItemId Bug</title>
        <script src="https://appsforoffice.microsoft.com/lib/1.1/hosted/office.js"></script>
    </head>
    <body class="bg-slate-100">
        <div id="out"></div>

        <script>
            Office.onReady(() => {
                const itemId = Office.context.mailbox.item.itemId
                document.getElementById('out').innerText = itemId

// if we add this (an empty handler), it works
                Office.context.mailbox.addHandlerAsync(
                    Office.EventType.ItemChanged,
                    () => {},
                    function (result) {
                        if (result.status !== Office.AsyncResultStatus.Succeeded) {
                            console.error(result.error)
                        }
                    }
                )

                Office.context.mailbox.addHandlerAsync(
                    Office.EventType.SelectedItemsChanged,
                    () => window.location.reload(),
                    function (result) {
                        if (result.status !== Office.AsyncResultStatus.Succeeded) {
                            console.error(result.error.message)
                        }
                    }
                )
            })
        </script>
    </body>
</html>
```

## Context
We can workaround this issue because of the mentioned workaround with the empty handler.

**Comment by kumarshrey-msft** [comment]
Can you try using [getSelectedItemsAsync](https://learn.microsoft.com/en-us/javascript/api/outlook/office.mailbox?view=outlook-js-preview#outlook-office-mailbox-getselecteditemsasync-member(1)) instead of the `Office.context.mailbox.item.itemId` API? You can refer to our [public documentation](https://learn.microsoft.com/en-us/office/dev/add-ins/outlook/item-multi-select?tabs=jsonmanifest#retrieve-the-subject-line-of-selected-messages) to see one of the sample use cases. You will no longer require the additional overhead of registering the ItemChanged event as well.

**Comment by Flyingdot** [comment]
Yes it does work. It seems to solve other kind of related issues like #4216 too. Downside is, that it requires API set 1.13, which excludes various important platforms according to the [docs](https://learn.microsoft.com/en-us/javascript/api/requirement-sets/outlook/outlook-api-requirement-sets?view=common-js-preview&tabs=jsonmanifest) (like on-prem, iOS, ...)

Also, it's different. `SelectedItemDetails` does not contain all the information `mailbox.item` does, for example categories.

**Comment by kumarshrey-msft** [comment]
Hi @Flyingdot, both [SelectedItemsChanged](https://learn.microsoft.com/en-us/javascript/api/office/office.eventtype?view=outlook-js-1.13&preserve-view=true) and [getSelectedItemsAsync](https://learn.microsoft.com/en-us/javascript/api/requirement-sets/outlook/requirement-set-1.13/office.context.mailbox?view=common-js-preview#methods) were released as part of [Requirement set 1.13](https://learn.microsoft.com/en-us/javascript/api/requirement-sets/outlook/requirement-set-1.13/outlook-requirement-set-1.13?view=common-js-preview) so you will need to support this as minimum supported API set.

Thank you for the input regarding `SelectedItemDetails`. This will be considered as a new feature request. We track Outlook add-in feature requests on our [Tech Community Page](http://aka.ms/M365dev-suggestions). You can submit your request there and choose the appropriate label(s). Feature requests on Tech Community are considered when we go through our planning process.
